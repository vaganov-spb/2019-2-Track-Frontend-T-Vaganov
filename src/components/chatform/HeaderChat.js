import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { HeaderTop } from '../chatslist/HeaderTop';
import { ArrowLeft } from './ArrowLeft';
import headStyles from '../../styles/ChatHeader.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

function HeaderChat(props) {
	return (
		<div className={headStyles.main_head}>
			<HeaderTop />
			<div className={headStyles.message_header_bottom}>
				<ArrowLeft>
					<Link style={{color: 'var(--white)',}} to="/">
						<i className="fa fa-arrow-left"/>
					</Link>
				</ArrowLeft>
				<div className={headStyles.message_header_img}>
					<img className={headStyles.avatar} alt="" src={props.src} />
				</div>

				<div className={headStyles.username}>
					<span className={headStyles.user}>{props.name}</span>
					<p className={headStyles.online}>в сети 2 часа назад</p>
				</div>
				<div className={headStyles.search_icons}>
					<span className={headStyles.right_icons}>
						<i className="fa fa-search" />
					</span>
					<span className={headStyles.right_icons}>
						<i className="fa fa-ellipsis-v" />
					</span>
				</div>
			</div>
		</div>
	);
}


const mapStateToProps = (state, ownProps) => {
	const chatId = ownProps.chatId;
	const obj = {
		chatId,
		name: state.chats.chats[chatId].name,
		src: state.chats.chats[chatId].url,
		isLoaded: state.chats.isLoaded,
		error: state.chats.error,
	};

	return obj;
};

export default connect(mapStateToProps)(HeaderChat);
