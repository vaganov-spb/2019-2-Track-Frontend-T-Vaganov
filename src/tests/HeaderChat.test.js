/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import HeaderChat from '../components/chatform/ChatHeaderInfo';

const mockStore = configureStore();

describe('HeaderChat renders and act correctly', () => {
	let props;
    let component;
    let store;

	beforeEach(() => {
		props = {
			chatId: 1,
        };
        
        store = mockStore({
				chats: {
					chats: {
                        1: {
                            name: 'test_name',
                            flag: true,
                            url: 'test_url',
                        }
                    },
                    error: null,
                    isLoaded: true,
				}
			});
        
		component = renderer
			.create(
                <Provider store={store}>
                    <MemoryRouter>
                        <HeaderChat { ...props } />
                    </MemoryRouter>  
                </Provider>
			)
			.toJSON();
         
	});

	it ('should render normally', () => {
		expect(component).toMatchSnapshot();
	});
});