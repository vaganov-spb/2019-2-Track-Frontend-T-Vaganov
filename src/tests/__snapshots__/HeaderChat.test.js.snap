// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderChat renders and act correctly should render normally 1`] = `
<div
  className="main_head"
>
  <div
    className="icons"
  >
    <div
      className="top_icons"
    >
      <div
        className="place_top_icons"
      >
        <span
          className="fa_icons"
        >
          <i
            className="fa fa-circle"
          />
        </span>
        <span
          className="fa_icons"
        >
          <i
            className="fa fa-battery-three-quarters"
          />
        </span>
        <span
          className="fa_icons"
        >
          <i
            className="fa fa-apple"
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="message_header_bottom"
  >
    <div
      className="arrow"
    >
      <span
        className="left_icon"
      >
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "var(--white)",
            }
          }
        >
          <i
            className="fa fa-arrow-left"
          />
        </a>
      </span>
    </div>
    <div
      className="message_header_img"
    >
      <img
        alt=""
        className="avatar"
        src="test_url"
      />
    </div>
    <div
      className="username"
    >
      <span
        className="user"
      >
        test_name
      </span>
      <p
        className="online"
      >
        в сети 2 часа назад
      </p>
    </div>
    <div
      className="search_icons"
    >
      <span
        className="right_icons"
      >
        <i
          className="fa fa-search"
        />
      </span>
      <span
        className="right_icons"
      >
        <i
          className="fa fa-ellipsis-v"
        />
      </span>
    </div>
  </div>
</div>
`;
